<?xml version="1.0" encoding="utf-8"?>
<launch>

  <!-- Load the URDF, SRDF and other .yaml configuration files on the param server -->
  <include ns="moveit_teleop" file="$(find iiwa_7_r800_moveit_config)/launch/planning_context.launch">
    <arg name="load_robot_description" value="true"/>
    <arg name="robot_description" value="iiwa_7_r800/robot_description"/>
  </include>

  <!-- GDB Debug Option -->
  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />
  <arg     if="$(arg debug)" name="launch_prefix"
       value="gdb -x $(find moveit_teleop)/launch/debug_settings.gdb --ex run --args" />

  <!-- Command line arguments -->
  <!-- <arg name="auto_step" default="false" /> -->
  <!-- <arg name="full_auto" default="true" /> -->
  <!-- <arg name="id" default="0" /> -->
  <!-- <arg name="fake_execution" default="0"/> -->
  <arg name="pose" default=""/>
  <arg name="play_traj" default="false"/>

  <!-- Planning Functionality -->
  <include ns="moveit_teleop" file="$(find iiwa_7_r800_moveit_config)/launch/planning_pipeline.launch.xml">
    <arg name="pipeline" value="ompl" />
  </include>

  <!-- Main process -->
  <node name="moveit_teleop" pkg="moveit_teleop" type="moveit_teleop_main" respawn="false"
    launch-prefix="$(arg launch_prefix)" output="screen"
    args="-play_traj=$(arg play_traj) --pose=$(arg pose)">
          <!-- - -full_auto=$(arg full_auto) - -auto_step=$(arg auto_step) -->
          <!-- - -id $(arg id)"> -->

    <!-- Robot-specific settings -->
    <rosparam command="load" file="$(find moveit_teleop)/config/config_iiwa_7_r800.yaml"/>
    <rosparam command="load" file="$(find iiwa_7_r800_moveit_config)/config/kinematics.yaml"/>

    <!-- Location-specific settings -->
    <rosparam command="load" file="$(find moveit_teleop)/config/collision_workspace.yaml"/>
  </node>

  <!-- <node name="game_controller" pkg="joy" type="joy_node" respawn="true"/> -->
</launch>
